server/src/domain/asset/asset.service.ts

```diff
@@ -187,26 +190,60 @@ export class AssetService {
         await this.access.requirePermission(authUser, Permission.ARCHIVE_READ, [dto.userId]);
       }
     }
+    if (dto.withPartners) {
+      if (
+        (dto.isArchived !== undefined && dto.isArchived !== false) ||
+        (dto.isTrashed !== undefined && dto.isTrashed !== false) ||
+        (dto.isFavorite !== undefined && dto.isFavorite !== false)
+      ) {
+        throw new BadRequestException(
+          'withPartners is only supported for non-archived, non-trashed, non-favorited assets',
+        );
+      }
+    }
   }
 
   async getTimeBuckets(authUser: AuthUserDto, dto: TimeBucketDto): Promise<TimeBucketResponseDto[]> {
     await this.timeBucketChecks(authUser, dto);
-    return this.assetRepository.getTimeBuckets(dto);
+    const timeBucketOptions = await this.buildTimeBucketOptions(authUser, dto);
+
+    return this.assetRepository.getTimeBuckets(timeBucketOptions);
   }
 
   async getTimeBucket(
     authUser: AuthUserDto,
     dto: TimeBucketAssetDto,
   ): Promise<AssetResponseDto[] | SanitizedAssetResponseDto[]> {
     await this.timeBucketChecks(authUser, dto);
-    const assets = await this.assetRepository.getTimeBucket(dto.timeBucket, dto);
+    const timeBucketOptions = await this.buildTimeBucketOptions(authUser, dto);
+    const assets = await this.assetRepository.getTimeBucket(dto.timeBucket, timeBucketOptions);
     if (authUser.isShowMetadata) {
       return assets.map((asset) => mapAsset(asset, { withStack: true }));
     } else {
       return assets.map((asset) => mapAsset(asset, { stripMetadata: true }));
     }
   }
 
+  async buildTimeBucketOptions(authUser: AuthUserDto, dto: TimeBucketDto): Promise<TimeBucketOptions> {
+    const { userId, ...options } = dto;
+    let userIds: string[] | undefined = undefined;
+
+    if (userId) {
+      userIds = [userId];
+
+      if (dto.withPartners) {
+        const partners = await this.partnerRepository.getAll(authUser.id);
+        const partnersIds = partners
+          .filter((partner) => partner.sharedBy && partner.sharedWith)
+          .filter((p) => p.inTimeline)
```

Comment:
Can't we merge those two filters into one by just adding`&& partner.inTimeline`. Doesn't matter too much performance-wise because there won't be that many partners. It is an extra (unnecessary) loop through the array though.